### DISCUSSION TITLE
How to disable plugin when "buf generate"

### DISCUSSION THREAD

**Discussion by boan-anbo** [bufbuild/buf#2912] [Q&A] [answered]
I have some plugins that should be **disabled** when generating `googleapis` when I run `npx buf generate buf.build/googleapis/googleapis --output output_path` but I don't know how to do it.

**Comment by doriable** [comment, marked as answer]
We currently do not have per-module configuration in `buf.gen.yaml`. For these cases, we recommend using multiple configuration files to manage plugins and inputs, for example `buf.googleapis.gen.yaml`, with the plugins you want for `googleapis`, and then running `npx buf generate buf.build/googleapis/googleapis --template buf.googleapis.gen.yaml`.
If you are using `v2` `buf.gen.yaml`, you can set `buf.build/googleapis/googleapis` in `buf.googleapis.gen.yaml` and so you won't need to pass it through as an argument (see docs for more details: https://buf.build/docs/configuration/v2/buf-gen-yaml#module)

**Reply by boan-anbo** [reply]
Thanks so much! Using --template is a much cleaner approach too.
